leku/src/main/java/com/schibstedspain/leku/search/LekuSearchTextWatcher.java

```diff
@@ -0,0 +1,36 @@
+package com.schibstedspain.leku.search;
+
+import android.text.Editable;
+import android.text.TextWatcher;
+
+abstract class LekuSearchTextWatcher implements TextWatcher {
+
+  private int minCharacters;
+
+  LekuSearchTextWatcher(int minCharacters) {
```

Comment:
As this implementation is used once, avoiding to make the characters a parameter could make the code simpler. Right now, we have to navigate a little to know what's the minimum characters, but at the same time, they will be always the same. So, I suggest to create a constant here instead of doing it at `LekuSearchFragment`.